src/entities/Basket/Basket.controller.ts

```diff
@@ -0,0 +1,20 @@
+import StatusCodes from 'http-status-codes';
+import { Request, Response } from 'express';
+import validateBasketReq from './Basket.validation';
+import Basket from './Basket.schema'
+
+const { BAD_REQUEST, CREATED, OK } = StatusCodes;
+
+export const addBasket = async (req: Request, res: Response) => {
+    const { error } = validateBasketReq(req.body);
+    if (error) return res.status(BAD_REQUEST).send(error.details[0].message);
+    const basketData = req.body;
+    const createdByUserId = {createdByUserId: req.user}
+    const basket = new Basket({...basketData, ...createdByUserId})
+    try {
+        await basket.save()
+        return res.status(CREATED).json(basket);
+    } catch (error) {
+        return res.status(BAD_REQUEST).send(error._message);
+    }
```

Comment:
Please check also if someone is not trying to sell the books to themselves :)
use User.findById(req.user). Then compare what you get with the recipient's id

Reply:
Done -> c4d7935dce33d5b805b6b7207a81df6ede492224